src/rust/terminusdb-community/src/graphql/schema.rs

```diff
@@ -435,22 +481,56 @@ impl<'a, C: QueryableContextType + 'a> GraphQLType for TerminusType<'a, C> {
     where
         DefaultScalarValue: 'r,
     {
-        let (name, frames) = info;
-        let frame = &frames.frames[name];
+        let name = &info.class;
+        let allframes = &info.allframes;
+        let frame = &allframes.frames[name];
         match frame {
             TypeDefinition::Class(d) => Self::generate_class_type(d, info, registry),
             TypeDefinition::Enum(_) => panic!("no enum expected here"),
         }
     }
 }
 
+fn rewind_rdf_list<'a>(instance: &'a dyn Layer, cons_id: u64) -> Option<u64> {
+    if let Some(rdf_rest) = instance.predicate_id(RDF_REST) {
+        let mut cons = Some(cons_id);
+        while let Some(id) = cons {
+            let res = instance
+                .triples_o(id)
+                .filter(|t| t.predicate == rdf_rest)
+                .map(|t| t.subject)
+                .next();
+            if res == None {
+                return cons;
+            } else {
+                cons = res
+            }
+        }
+        cons
+    } else {
+        None
+    }
+}
+
+fn subject_has_type<'a>(instance: &'a dyn Layer, subject_id: u64, class: &str) -> bool {
+    if let Some(rdf_type_id) = instance.predicate_id(RDF_TYPE) {
+        if let Some(class_id) = instance.object_node_id(class) {
+            instance.triple_exists(subject_id, rdf_type_id, class_id)
```

Comment:
if you return here you can save on two return falses by making it the default at the end.